voting/test/it/driver.js

```diff
@@ -164,59 +173,105 @@ class ElectionContracts {
 
+        expect(await contract.transact(signer, "unsafetests_setElectionMirrorPeriodInSeconds", Orbs.argUint64(this.options.votingMirrorPeriodInSeconds || 3))).to.be.successful;
+        expect(await contract.transact(signer, "unsafetests_setElectionVotePeriodInSeconds", Orbs.argUint64(this.options.votingValidityPeriodInSeconds || 140))).to.be.successful;
+        expect(await contract.transact(signer, "unsafetests_setElectionPeriodInSeconds", Orbs.argUint64(this.options.electionsPeriodInSeconds || 20))).to.be.successful;
     }
 
     async getOrbsValidatorAddresses() {
         const validators = await this.validators.getValidators();
         return Promise.all(validators.map(vAddr => this.validatorsRegistry.getOrbsAddress(vAddr)));
     }
 
+    async getOrbsValidatorReward(addr) {
+        let result = await this.orbs.contract(this.orbsVotingContractName).query(this.orbs.accounts[0], "getCumulativeValidatorReward", Orbs.argBytes(addr));
+        return Number(result.outputArguments[0].value)
+    }
 
     async setElectionBlockNumber() {
         const currentBlock = await this.ethereum.getLatestBlock();
         const electionBlock = currentBlock.number + 1;
 
-        await this.orbs.contract(this.orbsVotingContractName).transact(this.orbs.accounts[0], "unsafetests_setElectedBlockNumber", Orbs.argUint64(electionBlock));
+        await this.orbs.contract(this.orbsVotingContractName).transact(this.orbs.accounts[0], "unsafetests_setCurrentElectedBlockNumber", Orbs.argUint64(electionBlock));
 
         console.log(`Next election block number set to ${electionBlock}`);
 
         return electionBlock;
     }
 
+    async setElectionTimeToCurrentEthereumTime() {
+        const currentBlock = await this.ethereum.getLatestBlock();
+        await this.ethereum.waitForBlock(currentBlock.number + 1);
+
+        const nextBlockForCalculatingElectionTime = await this.ethereum.getLatestBlock();
+        const blockTimeInSeconds = nextBlockForCalculatingElectionTime.timestamp;
+
+        await this.orbs.contract(this.orbsVotingContractName).transact(this.orbs.accounts[0], "startTimeBasedElections");
+        expect(await this.orbs.contract(this.orbsVotingContractName).transact(this.orbs.accounts[0], "unsafetests_setCurrentElectionTimeNanos", Orbs.argUint64(blockTimeInSeconds * nanos))).to.be.successful;
+
+        console.log(`Next block number set to ${nextBlockForCalculatingElectionTime.number}`);
+
+        return nextBlockForCalculatingElectionTime.number;
+    }
+
+    async waitForMinimumBlocksAndPeriod() {
+        const currentBlock = await this.ethereum.getLatestBlock();
+        if (currentBlock.number < minBlocks) {
+            console.log(`\nEthereum interface doesn't have enough fake history, generating some...`);
+            await this.waitForOrbsFinality(minBlocks-currentBlock.number);
+        }
+    }
+
     async waitForOrbsFinality(blockToWaitFor) {
```

Comment:
I think this function should move to `Psilo`, and maybe later we should add some tests to it. It has nothing that's specific to Voting, right?

Reply:
it does need the finality .... do we want all the finality to be in psilo ? how ?

Reply:
If finality is a common concern (which I think it is) and is handled the same in all contracts, it can safely move to Psilo. But maybe in a separate PR.